R/download_ts.R

```diff
@@ -1,47 +1,56 @@
-#'@title download timeseries data to local file destination
-#'@description download a timeseries file to a user-specified (or temp file)
-#'  location
-#'@param site a valid mda.streams site (see \link{get_sites})
-#'@param var_src a valid variable name for timeseries data (see
-#'  \code{get_var_codes(out='var_src', type='ts')})
-#'@param folder string for a folder location
-#'@param ... additional arguments passed to
-#'  \code{\link[sbtools]{session_check_reauth}}
-#'  
-#'@return file handle for downloaded file
-#'@author Corinna Gries, Jordan S Read, Luke A Winslow
-#'@examples
-#'\dontrun{
-#'download_ts(site = 'nwis_06893300', var_src = 'doobs_nwis')
-#'}
-#'@import sbtools
-#'@import tools
-#'@export
-download_ts=function(site, var_src, folder = tempdir(), ...){
+#' @title download timeseries data to local file destination
+#' @description download a timeseries file to a user-specified (or temp file) 
+#'   location
+#' @param site a valid mda.streams site (see \link{get_sites})
+#' @param var_src a valid variable name for timeseries data (see 
+#'   \code{get_var_codes(out='var_src', type='ts')})
+#' @param folder string for a folder location
+#' @param on_exists character indicating what to do if the folder already 
+#'   contains a file with the intended download name
+#' @param ... additional arguments passed to 
+#'   \code{\link[sbtools]{session_check_reauth}}
+#'   
+#' @return file handle for downloaded file
+#' @author Corinna Gries, Jordan S Read, Luke A Winslow
+#' @examples
+#' \dontrun{
+#' download_ts(site = 'nwis_06893300', var_src = 'doobs_nwis')
+#' }
+#' @import sbtools
+#' @import tools
+#' @export
+download_ts=function(site, var_src, folder = tempdir(), on_exists=c("stop","skip","replace"), ...){
 
-  ts_variable <- paste0(pkg.env$ts_prefix, var_src)
-  scheme = get_scheme()
+  # check inputs
+  on_exists <- match.arg(on_exists)
```

Comment:
what kind of weird fail are you getting? it seems to be working for me, and defaulting (as i think it should, according to ?match.arg) to "stop". like this:
```
> library(mda.streams)
> download_ts(site = 'nwis_06893300', var_src = 'doobs_nwis')
[authentication]
[1] NA
> download_ts(site = 'nwis_01036390', var_src = 'doobs_nwis')
[1] "C:\\Users\\aappling\\AppData\\Local\\Temp\\1\\RtmpaqI53h/nwis_01036390-ts_doobs_nwis.tsv.gz"
> download_ts(site = 'nwis_01036390', var_src = 'doobs_nwis')
Error in download_ts(site = "nwis_01036390", var_src = "doobs_nwis") : 
  download destination already has a file and on_exists=='stop'
```
(caveat: i've been fiddling with other parts of this function this morning, so behavior might not quite match the pull request version. but i didn't change the `match.arg` line)